dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py

```diff
@@ -0,0 +1,276 @@
+                    urn = "urn:host:/{}".format(displayName)
+                identifiers.append(urn)
+                # derive useful labels and get labels from dynatrace tags
+                labels = self.get_labels(item)
+                labels += self.get_labels_from_dynatrace_tags(item)
+                if "tags" in item:
+                    del item["tags"]
+                # prefix the labels with `dynatrace-` for all labels
+                labels = ["dynatrace-{}".format(label) for label in labels]
+                data.update(item)
+                data.update({
+                    "identifiers": identifiers,
+                    "tags": self.tags,
+                    "domain": self.domain,
+                    "environment": self.environment,
+                    "instance": self.url,
+                    "labels": labels
+                })
+                data = self.filter_data(data)
+                self.component(externalId, component_type, data)
+                self.collect_relations(item, externalId)
+        else:
+            raise DynatraceError(response["error"].get("message"), response["error"].get("code"), component_type)
+
+    def get_labels_from_dynatrace_tags(self, item):
+        """
+        Process each tag as a label in component
+        :param item: the component item to read from
+        :return: list of added tags as labels
+        """
+        tags = []
+        for tag in item.get("tags", []):
+            tag_label = ''
+            if bool(tag['context']) and tag.get('context') != 'CONTEXTLESS':
+                tag_label += '[{0}]'.format(tag['context'])
+            if bool(tag.get('key')):
+                tag_label += "{0}".format(tag['key'])
+            if bool(tag.get('value')):
+                tag_label += ":{0}".format(tag['value'])
+            tags.append(tag_label)
+        return tags
+
+    def get_labels(self, item):
+        """
+        get labels for each component
+        :param item: the component item
+        :return: the list of added labels for a component
+        """
+        labels = []
+        # append management zones in labels for each existing component
+        if "managementZones" in item:
+            for zone in item["managementZones"]:
+                labels.append("managementZones:{}".format(zone.get("name")))
+        if "entityId" in item:
+            labels.append(item["entityId"])
+        if "monitoringState" in item:
+            actual_state = item["monitoringState"].get("actualMonitoringState")
+            expected_state = item["monitoringState"].get("expectedMonitoringState")
+            labels.append("actualMonitoringState:{}".format(actual_state))
+            labels.append("expectedMonitoringState:{}".format(expected_state))
+        if "softwareTechnologies" in item:
+            for technologies in item["softwareTechnologies"]:
+                tech_label = ''
+                if bool(technologies['type']):
```

Comment:
Why do you add this `bool` around it? That should do nothing right?  

Also, perhaps use `.get('type')` here and below instead? I don't think we should want to fail on this.